### ISSUE TITLE
unset RANGER_LEVEL when opening new terminal

### ISSUE THREAD

**Issue by FlyingWombat** [ranger/ranger#1461]
#### TL;DR  
Terminals spawned by `:terminal` inherit RANGER_LEVEL.  
I think default behavior should be that RANGER_LEVEL is unset, when spawning a new terminal.

#### Runtime Environment

- Operating system and version: Arch Linux x86_64 v4.20.4; zsh 5.6.2
- Terminal emulator and version: termite v14
- Python version: 3.7.2
- Ranger version/commit: 66f0a462
- Locale: en_US.UTF-8

#### Current Behavior
As is normal, the new terminal inherits env from ranger, which includes RANGER_LEVEL.

#### Expected Behavior
Since the new terminal is not related to the ranger instance that spawned it -- ranger is not one of its parents -- RANGER_LEVEL should be unset.  
I brought this up, because I think this should be default behavior, but am unsure about the best way to do that. To clarify, I think it should only be unset in cases where the spawned process is going to be disowned (such as the `:terminal` command).  
If there's a good reason to keep current behavior, that's no problem. It's trivial to override it in commands.py  
See Possible Solutions, below.

#### Context

This will only affect someone, if they have programs that check for RANGER_LEVEL.  
For example: if you have the following in your shell config, you cannot start ranger in a terminal spawned by ranger.  
``` bash
# prevent nested ranger instances
ranger() {
	if [ -z "$RANGER_LEVEL" ]; then
		env TERMCMD=termite /usr/bin/ranger "$@"
	else
		exit
	fi
}
```

#### Steps to reproduce
1. open `ranger`
2. run `:terminal`
3. run `env | grep RANGER_LEVEL`, in that new terminal

#### Possible Solutions
One could simply override the `:terminal` command to unset RANGER_LEVEL.  
From my commands.py:
``` python
class terminal(Command):  # pylint: disable = invalid-name
    """:terminal
    Spawns an "x-terminal-emulator" starting in the current directory.
    And removes RANGER_LEVEL from env, for the new terminal.
    """
    def execute(self):
        from ranger.ext.get_executables import get_term
        self.fm.run(get_term() + """ -e "$SHELL -c 'unset -v RANGER_LEVEL && $SHELL'" """, flags='f')
```

**Comment by toonn** [comment]
You're absolutely right. This patch would definitely be upstreamed.

**Comment by palikar** [comment]
@FlyingWombat are you planning to do a PR with your solution?

**Comment by FlyingWombat** [comment]
Oh, ya, for some reason I thought @toonn would do it; I can do that.  

First, there are one small thing I would like to discuss:  
I've discovered that there maybe trouble with differences in how different terminal emulators handle quotes in their parsers.  
As an example: this command -- `termite -e "$SHELL -c 'ls && $SHELL'"` -- works for termite and xterm, but not alacritty.
`alacritty -e $SHELL -c "ls && $SHELL"` works for alacritty and xterm, but not termite.  
For either that doesn't work the error is something like `ranger && /bin/zsh : no such file or directory`
I can't find any quoting scheme that works for all three.

The underlying problem for this, is the shell doesn't natively support a way to pre-load commands into an interactive session.

**Comment by toonn** [comment]
Oh, this means it's dependent on the implementation of `flag t` which is changing. So I'll take this into account when I find the time to work on that.

**Comment by FlyingWombat** [comment]
I did some more testing on various terminals:  
1. `-e "$SHELL -c 'ls && $SHELL'"`
2. `-e $SHELL -c "ls && $SHELL"`

Works for 1 only:  termite, sakura, gnome-terminal  
Works for 2 only: alacritty, urxvt, kitty  
Works for both: xterm, konsole, lxterminal, tilix  
Doesn't work for either: qterminal

**Comment by toonn** [comment]
Check #1108 for some compatibility across terminal checking I've done.
